manim/utils/decorators.py

```diff
@@ -0,0 +1,445 @@
+"""Function decorators."""
+
+__all__ = ["deprecated", "deprecated_params"]
+
+
+import re
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+from decorator import decorate, decorator
+
+from .. import logger
+
+
+def get_callable_info(callable: Callable) -> Tuple[str, str]:
+    """Returns type and name of a callable.
+
+    Parameters
+    ----------
+    callable
+        The callable
+
+    Returns
+    -------
+    Tuple[str, str]
+        The type and name of the callable. Type can can be one of "class", "method" (for
+        functions defined in classes) or "function"). For methods, name is Class.method.
+    """
+    what = type(callable).__name__
+    name = callable.__qualname__
+    if what == "function" and name[0].isupper():  # TODO a bit hacky but works
+        what = "method"
+    elif what == "type":
+        what = "class"
+    return (what, name)
+
+
+def deprecation_text_component(
+    since: Optional[str], until: Optional[str], message: str
```

Comment:
[`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) should only be used when the default value is `None`, this isn't the case here.

Reply:
https://docs.python.org/3/library/typing.html says:
> You can use `Optional[X]` as a shorthand for `Union[X, None]`.

That is what I do here.